### DISCUSSION TITLE
improve robustness of post-pve-install.sh apt option

### DISCUSSION THREAD

**Discussion by github-tomster** [tteck/Proxmox#2142] [Ideas]
hello world

the script can call:
```
apt-get update &>/dev/null
apt-get -y dist-upgrade &>/dev/null
```
this means no apt-related messages that require an user-interaction are displayed to the screen.
in case there are messages - for example grub related ones - the procedure waits forever ⏳

is it by design or can this script only run on fresh installs?

**Comment by tteck** [comment]
The script is specifically focused on source lists and minor tweaks, it's not intended to handle broader issues or functionalities outside of its designated scope. Grub related issues?

**Comment by github-tomster** [comment]
hence mentioning the apt option.
i have had a machine - unfortunately not in reach anymore - the `post-pve-install.sh` stood at `"Updating Proxmox VE (Patience)"`
back on the screen, a few hours later, i checked the apt.log and found some grub related questions about partitions and kernels [i do not remember exactly]
it seems that the case a question is printed while doing `apt-get -y dist-upgrade &>/dev/null` there is no chance to interact.
or to say it in an other way:
running the default  `post-pve-install.sh` and choosing `Y` for `Update Proxmox VE now?` no interaction of any kind is possible.
maybe remove the apt option an keep focus on source&tweaks 🤷🏻 
or do not `>/dev/null` the apt part 🤷🏻 
i understand that this case - my case - is considered very-rare, but not excluded 🤡

**Reply by tteck** [reply]
Yes, your case is considered very rare and is excluded. I apologize for any inconvenience.

**Reply by tteck** [reply]
Why don't you just execute
```
apt-get update && apt-get -y dist-upgrade
```
sounds like that's what you're using the script for anyway.

**Reply by github-tomster** [reply]
no harm done
i am very glad to demonstrate that these cases are out there and maybe they could be taken into account or others find this helpful to not going desperate ...

edit:
i do `apt` not exclusively with `post-pve-install.sh` i did it by chance and had such a lucky strike ...
